exchanges/kraken/kraken_websocket.go

```diff
@@ -148,28 +151,16 @@ func (k *Kraken) WsHandleData() {
 }
 
 // wsPingHandler sends a message "ping" every 27 to maintain the connection to the websocket
-func (k *Kraken) wsPingHandler() {
-	k.Websocket.Wg.Add(1)
-	defer k.Websocket.Wg.Done()
-	ticker := time.NewTicker(time.Second * 27)
-	defer ticker.Stop()
-
-	for {
-		select {
-		case <-k.Websocket.ShutdownC:
-			return
-		case <-ticker.C:
-			pingEvent := WebsocketBaseEventRequest{Event: krakenWsPing}
-			if k.Verbose {
-				log.Debugf(log.ExchangeSys, "%v sending ping",
-					k.Name)
-			}
-			err := k.WebsocketConn.SendMessage(pingEvent)
-			if err != nil {
-				k.Websocket.DataHandler <- err
-			}
-		}
+func (k *Kraken) wsPingHandler() error {
+	message, err := json.Marshal(pingRequest)
+	if err != nil {
+		return err
 	}
+	k.WebsocketConn.SetupPingHandler(wshandler.WebsocketPingHandler{
```

Comment:
should set message type here

Reply:
Just reporting that the latest commit fixes the [ERROR] | 03/01/2020 11:23:24 | Kraken failed to send message to websocket {"event":"ping"} error